### PR TITLE
Demo test for exchanger

### PR DESCRIPTION
**Pull request by Mingela**
Added integration test demonstrating exchanger work as a dApp-contract

### CODE REVIEW

**Review by yurivin — APPROVED** [review]

File: dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt [hunk 1]
```diff
@@ -152,56 +164,59 @@ class DappTestEnvironment : Closeable {
...
-        )
-
-        chainAdapter.addFileSystemBind(
-            resourcesLocation,
-            "/opt/chain-adapter/configs",
-            BindMode.READ_WRITE
-        )
-
-        chainAdapter.withNetwork(iroha.network).start()
+        chainAdapter.withNetwork(iroha.network)
```
Comment (dolgopolovwork) [inline dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt:176]: Do you close this container?

File: dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt [hunk 2]
```diff
@@ -140,8 +151,9 @@ class DappTestEnvironment : Closeable {
         iroha.configure()
         postgresDockerContainer = iroha.postgresDockerContainer
         postgresDockerContainer.start()
-        irohaContainer = iroha.irohaDockerContainer.withCreateContainerCmdModifier { it.withName("iroha") }
-            .withExposedPorts(DEFAULT_IROHA_PORT)
+        irohaContainer =
```
Comment (dolgopolovwork) [inline dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt:153]: Don't forget to close.

File: dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt [hunk 3]
```diff
@@ -152,56 +164,59 @@ class DappTestEnvironment : Closeable {
...
+            )
+            .withEnv(
+                "WAIT_HOSTS",
+                "$irohaContainerName:$irohaPort, $rmqContainerName:$rmqPort"
+            )
+            .start()
 
         queryAPI = QueryAPI(irohaAPI, dappInstanceAccountId, irohaKeyPair)
 
         val chainListener = ReliableIrohaChainListener(
```
Comment (dolgopolovwork) [inline dapp-core/src/test/kotlin/jp/co/soramitsu/dapp/test/environment/DappTestEnvironment.kt:209]: Do you close it?

File: dapp-core/src/test/resources/exchanger_contract.groovy [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+        for (int i = 0; i < ITERATIONS; i++) {
+            fistSum += compute(sourceAssetBalance, targetAssetBalance, heightStep * i + heightStep / 2)
+            secondSum += compute(sourceAssetBalance, targetAssetBalance, heightStep * i)
+        }
+
+        return heightStep / 6 * (
+                compute(sourceAssetBalance, targetAssetBalance, 0) +
+                        compute(sourceAssetBalance, targetAssetBalance, amount) +
+                        4 * fistSum + 2 * secondSum
```
Comment (dolgopolovwork) [inline dapp-core/src/test/resources/exchanger_contract.groovy:44]: What are all these numbers? Create constants please

File: dapp-core/src/test/resources/exchanger_contract.groovy [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        )
+    }
+
+    @Override
+    Iterable<Commands.Command.CommandCase> getCommandsToMonitor() {
+        return [Commands.Command.CommandCase.TRANSFER_ASSET]
+    }
+
+    @Override
+    synchronized void processCommand(Commands.Command command, long createdTime) {
```
Comment (dolgopolovwork) [inline dapp-core/src/test/resources/exchanger_contract.groovy:54]: Why `synchronized`?

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**